packages/url_launcher/url_launcher_macos/example/macos/RunnerTests/RunnerTests.swift

```diff
@@ -6,19 +6,106 @@ import FlutterMacOS
 import XCTest
 import url_launcher_macos
 
+class StubWorkspace: URLLauncher {
+  var isSuccessful = true
+  var url = URL.init(string: "https://flutter.dev")
+
+  func open(_ url: URL) -> Bool {
+    return isSuccessful
+  }
+
+  func urlForApplication(toOpen: URL) -> URL? {
+    return url
+  }
+
+}
+
 class RunnerTests: XCTestCase {
-  func testCanLaunch() throws {
-    let plugin = UrlLauncherPlugin()
+
+  var workspace: StubWorkspace! = nil
+  var pluginWithStubWorkspace: UrlLauncherPlugin! = nil
+  var plugin: UrlLauncherPlugin! = nil
+
+  override func setUp() {
+    workspace = StubWorkspace()
+    pluginWithStubWorkspace = UrlLauncherPlugin(workspace)
+    plugin = UrlLauncherPlugin()
+  }
+
+  override func tearDown() {
+    workspace = nil
+    pluginWithStubWorkspace = nil
+    plugin = nil
+  }
+
+  func testCanLaunchSuccessTrue() throws {
     let call = FlutterMethodCall(
       methodName: "canLaunch",
       arguments: ["url": "https://flutter.dev"])
-    var canLaunch: Bool?
+
+    plugin.handle(
+      call,
+      result: { (result: Any?) -> Void in
+        XCTAssertNotNil(result)
+        let canLaunch = result! as? Bool
+        XCTAssertNotNil(canLaunch)
+        XCTAssertTrue(canLaunch!)
+      })
+  }
+
+  func testCanLaunchFailureFalse() throws {
```

Comment:
I'm not sure what these names are meant to indicate exactly. What is the distinction between "Success/Failure" and "True/False"? The fact that you only have "SuccessTrue" and "FailureFalse" suggests that there isn't one.

Reply:
renamed according to the tested conditions.